### DISCUSSION TITLE
RTK Query: Provider with a context instance

### DISCUSSION THREAD

**Discussion by craigrbarnes** [reduxjs/redux-toolkit#2123] [Q&A] [answered]
I was migrating some code over from react-toolkit to RTK Query. I got this error: 

`Error: could not find react-redux context value; please ensure the component is wrapped in a <Provider>`

After some investigating I figured out that my Provider is also supplying a context instance:
```
 <Provider store={coreStore} context={CoreContext}>
```
which explains the error. Removing the context prop eliminated the error. Is there a way to use a context instance with RTK Query?

**Comment by phryneas** [comment, marked as answer]
Yes, by using `buildCreateApi` and putting your context-bound hooks in there:
```js
const createApi = buildCreateApi(
  coreModule(),
  reactHooksModule({useSelector, useDispatch, useStore})
)
```

**Comment by markerikson** [comment]
Out of curiosity, _why_ are you using a custom context instance with React-Redux?

**Reply by craigrbarnes** [reply]
We are designing and maybe it is overkill, a multilevel store so what we apps, which are analytics tools built with react, can have their own store/context which is scoped to within the app.
